src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -971,16 +963,264 @@ void CmdSketcherViewSection::activated(int iMsg)
 
 bool CmdSketcherViewSection::isActive()
 {
-    Gui::Document *doc = getActiveGuiDocument();
-    if (doc) {
-        // checks if a Sketch Viewprovider is in Edit and is in no special mode
-        SketcherGui::ViewProviderSketch* vp = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
-        if (vp /*&& vp->getSketchMode() == ViewProviderSketch::STATUS_NONE*/)
-            return true;
+    return isSketchInEdit(getActiveGuiDocument());
+}
+
+/* Grid tool */
+class GridSpaceAction : public QWidgetAction
+{
+public:
+    GridSpaceAction(QObject* parent) : QWidgetAction(parent) {
+    }
+
+    void updateWidget() {
+
+        auto* sketchView = getView();
+
+        if(sketchView) {
+
+            auto updateCheckBox = [](QCheckBox * checkbox, bool value) {
+                auto checked = checkbox->checkState() == Qt::Checked;
+
+                if( value != checked ) {
+                    const QSignalBlocker blocker(checkbox);
+                    checkbox->setChecked(value);
+                }
+            };
+
+            auto updateCheckBoxFromProperty = [updateCheckBox](QCheckBox * checkbox, App::PropertyBool & property) {
+                auto propvalue = property.getValue();
+
+                updateCheckBox(checkbox, propvalue);
+            };
+
+            updateCheckBox(gridSnap, sketchView->getSnapMode() == SnapMode::SnapToGrid);
+
+            updateCheckBoxFromProperty(gridAutoSpacing, sketchView->GridAuto);
+
+            auto autospacing = gridAutoSpacing->checkState() == Qt::Checked;
+
+            gridSizeBox->setEnabled(!autospacing);
+
+            gridSizeBox->setValue(sketchView->GridSize.getValue());
+        }
+    }
+
+    void languageChange()
+    {
+        gridSnap->setText(QApplication::translate("GridSpaceAction", "Grid Snap"));

```

Comment:
Why not do simply :
```suggestion
        gridSnap->setText(tr("Grid Snap"));

```
Same for roughly all translations

Reply:
Yes. Translations were changed over from place to place, and I think it now makes sense to just use the class name as context and have the code cleaner.

Reply:
I just have a doubt here. IMO, we have to add the `Q_OBJECT` macro to the class or it will use the `QWidgetAction` tr functions, and so context will be set to "QWidgetAction". Not 100% sure, but I think it is how it works.